Don’t compare non-enumerable symbols

Since version 4.0.1 (67d704ca19ff9e1acd7ddea9fbf7a4157917921a / #81, CC @snewcomer and @keithamus), deep-eql compares non-enumerable symbols. I find this surprising and unintuitive, and it’s different from string keys, where only enumerable keys are compared:

```js
> key = 'x';
> deepEql(Object.defineProperty({}, key, { value: 'x', enumerable: false }), Object.defineProperty({}, key, { value: 'y', enumerable: false }))
true
> key = Symbol.for('x');
> deepEql(Object.defineProperty({}, key, { value: 'x', enumerable: false }), Object.defineProperty({}, key, { value: 'y', enumerable: false }))
false
```

@meeber in https://github.com/chaijs/chai/issues/1054#issuecomment-331317911 also only suggested that *enumerable* symbols should be compared, and I suggest changing deep-eql to only compare these.

To provide some context for the “unintuitive” description, if you’re curious: in [m3api-query](https://github.com/lucaswerkmeister/m3api-query/), I use non-enumerable symbol properties to attach additional information to objects that users of my library might find useful, but won’t necessarily need all of the time; I don’t need them to show up in `console.log()`, for example. (In fact, I was under the mistaken impression that, without a reference to the symbol, you couldn’t access the property at all: I was unaware that `Object.getOwnPropertyNames()` and `Object.getOwnPropertySymbols()` include non-enumerable properties in their return values.) In tests that aren’t focused on these additional properties, I compare the objects returned by my library with object literals that only have the “main” properties, and expect those comparisons to succeed; after an `npm update`, they failed.

The assertion failure message was also not easy to understand, unfortunately:
```
      AssertionError: expected { Object (revid) } to deeply equal { revid: 123 }                                                                   
      + expected - actual                                                                                                                          
```
But that’s probably not deep-eql’s fault, I think the message formatting is somewhere else in Chai.

--- a/index.js
+++ b/index.js
@@ -392,8 +392,15 @@ function getEnumerableKeys(target) {
   return keys;
 }
 
-function getNonEnumerableSymbols(target) {
-  var keys = Object.getOwnPropertySymbols(target);
+function getEnumerableSymbols(target) {
+  var keys = [];
+  var allKeys = Object.getOwnPropertySymbols(target);
+  for (var i = 0; i < allKeys.length; i += 1) {
+    var key = allKeys[i];
+    if (Object.getOwnPropertyDescriptor(target, key).enumerable) {
+      keys.push(key);
+    }
+  }
   return keys;
 }
 
@@ -432,8 +439,8 @@ function keysEqual(leftHandOperand, rightHandOperand, keys, options) {
 function objectEqual(leftHandOperand, rightHandOperand, options) {
   var leftHandKeys = getEnumerableKeys(leftHandOperand);
   var rightHandKeys = getEnumerableKeys(rightHandOperand);
-  var leftHandSymbols = getNonEnumerableSymbols(leftHandOperand);
-  var rightHandSymbols = getNonEnumerableSymbols(rightHandOperand);
+  var leftHandSymbols = getEnumerableSymbols(leftHandOperand);
+  var rightHandSymbols = getEnumerableSymbols(rightHandOperand);
   leftHandKeys = leftHandKeys.concat(leftHandSymbols);
   rightHandKeys = rightHandKeys.concat(rightHandSymbols);
 

--- a/test/index.js
+++ b/test/index.js
@@ -444,6 +444,16 @@ describe('Generic', function () {
       objectB = { [symb3]: 'a', [symb2]: 'b' };
       assert(eql(objectA, objectB) === true, 'eql(obj, obj)');
     });
+
+    it('ignores non-enumerable symbols', function () {
+      var symb = Symbol('a');
+      var symb2 = Symbol('b');
+      var objectA = { [symb]: 'a' };
+      Object.defineProperty(objectA, symb2, { value: 'b', enumerable: false });
+      var objectB = { [symb]: 'a' };
+      Object.defineProperty(objectB, symb2, { value: 'c', enumerable: false });
+      assert(eql(objectA, objectB) === true, 'eql(obj, obj)');
+    });
   });
 
 

